pyiceberg/table/snapshots.py

```diff
@@ -228,6 +229,13 @@ def __eq__(self, other: Any) -> bool:
         )
 
 
+@lru_cache
```

Comment:
what are your thoughts on this, wouldn't be clearer if we could explicitly pass the max size for the lru_cache annotation, to store max entries rather than going default.
`@lru_cache(maxsize=128)`

Reply:
As [the default is 128](https://docs.python.org/3/library/functools.html#functools.lru_cache), I don't think we need to explicitly define `maxsize=128`.

Also, `_manifests()` is not a public API, so we would *probably* not set the cache size through user config via pyiceberg.yaml. Is there a use-case where the user would want to set the cache size?

Reply:
Agree the default size is what you mentioned, It was just an example, but the idea was to make it more readable.
Please feel free to take your call.

Reply:
is 128 big enough?

Reply:
I think 128 should be enough. We're caching `manifest_list` here and the number of manifest lists should be <100 in most cases. 
We can always make this configurable in the future :)